### DISCUSSION TITLE
using pyinstaller when importing pygsheets

### DISCUSSION THREAD

**Discussion by danagold404** [nithinmurali/pygsheets#489] [Q&A] [answered]
Hi everyone!
I'm trying to create an exe file to distribute among users in my company.
the exe file open a simple GUI (made with Tkinter), which allows the user to select a txt file from their pc and inserts the data from the txt into a Google sheet. 
I tested my python script before attempting to turn it into an exe and it worked great.
but now, when I'm trying to use Pyinstaller it shows the following error:

```
C:\Users\Asus\PycharmProjects\dmtrial\venv\dist>DMPulse2GS.exe
Traceback (most recent call last):
  File "pygsheets\sheet.py", line 39, in __init__
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Asus\\AppData\\Local\\Temp\\_MEI102362\\pygsheets\\data\\sheets_discovery.json'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "DMPulse2GS.py", line 39, in <module>
  File "pygsheets\authorization.py", line 131, in authorize
  File "pygsheets\client.py", line 61, in __init__
  File "pygsheets\sheet.py", line 42, in __init__
  File "googleapiclient\_helpers.py", line 134, in positional_wrapper
  File "googleapiclient\discovery.py", line 291, in build
  File "googleapiclient\discovery.py", line 405, in _retrieve_discovery_doc
googleapiclient.errors.UnknownApiNameOrVersion: name: sheets  version: v4
[11284] Failed to execute script DMPulse2GS
```

this is the cmd command I used to create the exe file:
`C:\Users\Asus\PycharmProjects\dmtrial\venv>pyinstaller --onefile DMPulse2GS.py`

it created three folders in this directory called: 'dist', 'build', and '__pycache__'

after it finished running successfully, I copied my Service Accounts Key (json file) to the 'dist' folder created by Pyinstaller. (I also tried adding the file using --add-data when running pyinstaller for the first time on the file but it returns the same error)

then I tried clicking the exe file, which was when it returned the above error.

I tried everything!! any help / tips would be amazing! Thanks in advance!

**Comment by nithinmurali** [comment]
for some reason your script cant download the discovery document. can you try adding it with --add-data? this one "C:\\Users\\Asus\\AppData\\Local\\Temp\\_MEI102362\\pygsheets\\data\\sheets_discovery.json"

**Comment by nithinmurali** [comment, marked as answer]
A fix for this is deployed in staging.  [more info](https://github.com/nithinmurali/pygsheets/issues/490#issuecomment-847714545)
